### ISSUE TITLE
Metabase finds empty schema on a MongoDB collection with actual data available.

### ISSUE THREAD

**Issue by ildella** [metabase/metabase#7420]
Hi. 
Is the second MongoDB I use metabase on a MongoDB. 
Both of the times, while most of the collections are scanned properly (I see all fields), a few are empty. There are no fields available in the Data Model section. 

This time is the Users collection. I've tried clean/rescan multiple times. 

I'm running metabase from docker. This is the log when I click on "discard cached field values" over the specific collection

```
04-24 14:24:04 DEBUG metabase.middleware :: POST /api/table/11/discard_values 200 (787 µs) (1 DB calls). Jetty threads: 8/50 (3 busy, 5 idle, 0 queued)
```
Then if I click on 're-scan this table'
```
04-24 14:25:19 DEBUG metabase.middleware :: POST /api/table/11/rescan_values 200 (984 µs) (0 DB calls). Jetty threads: 8/50 (3 busy, 5 idle, 0 queued)
[...]
04-24 14:50:00 DEBUG sync.util :: Sync operations in flight: {:sync-metadata #{2}}
04-24 14:50:00 INFO sync.util :: STARTING: Sync metadata for mongo Database 2 'my-database'
04-24 14:50:15 ERROR sync.util :: Error syncing fields for Table 11 'users': class java.lang.NullPointerException
[]
...
```

The DB has just been connected. All permissions are green on all collections. 
The DB have ~30 collections and ~5 have an empty schema detected by Metabase.

Any tip? 
Thanks. 

- Your browser and the version: Firefox 50
- Your operating system: Ubuntu 18.10
- mongodb 3.4
- Metabase version: 0.28.5
- Metabase hosting environment: Ubuntu 17.10
- Metabase internal database: H2 (default)

**Comment by salsakran** [comment]
can you share the schema of the collections that are empty? sounds like it is hitting some kind of error as it scans and not saving metadata for that collection.

**Comment by gkoberger** [comment]
I think I'm having this same problem. I'm on `v0.29.0` (and mongo version 3.4.14) and most of my schemas are completely empty. I've tried rescanning a bunch, with no luck. I don't see any errors in the logs, though.

How does scanning with Mongo work? Is there something on my end I might be doing wrong?

Here's my logs when I hit rescan:

```
    DEBUG metabase.middleware :: POST /api/table/23/rescan_values 200 (1 ms) (1 DB calls). Jetty threads: 8/50 (3 busy, 5 idle, 0 queued)
```

Thanks!

**Comment by jornh** [comment]
There is another related issue - with a proposed workaround https://github.com/metabase/metabase/issues/6975#issuecomment-377115094 (A fair warning you want to try out the work-around in a non-production environment).

I think it would be good if you guys @ildella and @gkoberger can either confirm or dismiss if this is the same underlying problem with your particular schemas and datasets.

**Comment by rjurado01** [comment]
I have similar problem with this collection: https://github.com/ozlerhakan/mongodb-json-files/blob/master/datasets/tweets.zip

**Comment by tschut** [comment]
Ran into the same issue yesterday. Already suspected null values, after reading #6975 am almost sure. Think this one is a duplicate, I'll add my comments on the other issue.

**Comment by camsaul** [comment]
Duplicate of #6975

Fixed by #8072
